api/handlers.go

```diff
@@ -794,26 +858,38 @@ func (api *API) userReconfirmPOST(w http.ResponseWriter, req *http.Request, _ ht
 	api.WriteSuccess(w)
 }
 
-// userRecoverGET allows the user to request an account recovery. This creates
-// a password-reset token that allows the user to change their password without
-// logging in.
+// userRecoverRequestPOST allows the user to request an account recovery. This
+// creates a password-reset token that allows the user to change their password
+// without logging in.
 // The user doesn't need to be logged in.
-func (api *API) userRecoverGET(w http.ResponseWriter, req *http.Request, _ httprouter.Params) {
-	if err := req.ParseForm(); err != nil {
+func (api *API) userRecoverRequestPOST(w http.ResponseWriter, req *http.Request, _ httprouter.Params) {
+	defer api.closeBody(req)
+	// Read and parse the request body.
+	bodyBytes, err := ioutil.ReadAll(req.Body)
+	if err != nil {
+		err = errors.AddContext(err, "failed to read request body")
 		api.WriteError(w, err, http.StatusBadRequest)
 		return
 	}
-	email := req.Form.Get("email")
-	if email == "" {
+	var payload struct {
+		Email string `json:"email"`
+	}
+	err = json.Unmarshal(bodyBytes, &payload)
+	if err != nil {
+		err = errors.AddContext(err, "failed to parse request body")
+		api.WriteError(w, err, http.StatusBadRequest)
+		return
+	}
+	if payload.Email == "" {
 		api.WriteError(w, errors.New("missing required parameter 'email'"), http.StatusBadRequest)
 		return
 	}
-	u, err := api.staticDB.UserByEmail(req.Context(), email)
+	u, err := api.staticDB.UserByEmail(req.Context(), payload.Email)
```

Comment:
here you're passing in the `Email` you received from the user directly without parsing the address. I think that's annoying UX right because on user creation or update you allow to pass value X (which internally gets transformed into X') but then when he wants to recover he would send X again, and you wouldn't find the user by email because you're not parsing it here. Right?

Reply:
Oh, nice catch! Yes, you are absolutely right. I'll add a helper method for cleaning the email and I'll use it everywhere.

Reply:
I think this is still an issue?

Reply:
No, this is no longer an issue because if they try to use any email which is not a valid and simple (e.g. joe@example.com) the registration will fail. So there is no need to validate here.

Reply:
oh ok so we updated registration! ok great thanks